### PR TITLE
Wrap `from_lists` and `from_flat`

### LINKED ISSUE #43: Add wrapper for Nested-Pandas `from_flat`
**Feature request**
We should add a wrapper for `from_flat` function which was just added to Nested-Pandas: https://github.com/lincc-frameworks/nested-pandas/pull/123

**Before submitting**
Please check the following:

- [x] I have described the purpose of the suggested change, specifying what I need the enhancement to accomplish, i.e. what problem it solves.
- [x] I have included any relevant links, screenshots, environment information, and data relevant to implementing the requested feature, as well as pseudocode for how I want to access the new functionality.
- [ ] If I have ideas for how the new feature could be implemented, I have provided explanations and/or pseudocode and/or task lists for the steps.

### PR DESCRIPTION
**Pull request by dougbrn**
## Change Description

Resolves #43 as well as adding a wrapper for from_lists. Updates the nested-pandas dep version to 0.2.1, so a v0.2.1 release of this should follow (skipping 0.2.0 to keep aligned with nested-pandas).
- [x] My PR includes a link to the issue that I am addressing

## Code Quality
- [x] I have read the Contribution Guide
- [x] My code follows the code style of this project
- [x] My code builds (or compiles) cleanly without any errors or warnings
- [x] My code contains relevant comments and necessary documentation

### CODE REVIEW

File: src/nested_dask/core.py [hunk 1]
```diff
@@ -102,7 +103,7 @@ def __setitem__(self, key, value):
                 new_flat = new_flat.astype({col: pd.ArrowDtype(pa.string())})
 
             # pack the modified df back into a nested column
-            meta = self._nested_meta_from_flat(new_flat, nested)
+            meta = _nested_meta_from_flat(new_flat, nested)
             packed = new_flat.map_partitions(lambda x: pack(x), meta=meta)
```
Comment (hombit) [inline src/nested_dask/core.py:107]: I'd recommend to use the same dtype we have in meta with `pack(dtype=)`

**Review by hombit — APPROVED** [review]
It looks nice! I have one general question, and one discussion topic.

Q: can we make meta with empty frames instead?

Discussion: what if we want patch nested-dask without patching nested-pandas? I'd propose syncronise minor version only, while keeping patch for bug fixes. It would also help as follow more semver-like model: when we change the interface of nested-pandas we should  change minor versions of both packages.

File: src/nested_dask/core.py (hunk 1 above)
Comment (dougbrn) [reply src/nested_dask/core.py]: Makes sense, implemented

**Comment by dougbrn** [comment]
> It looks nice! I have one general question, and one discussion topic.
> 
> Q: can we make meta with empty frames instead?

Do you mean as opposed to using the `_meta` property?
> 
> Discussion: what if we want patch nested-dask without patching nested-pandas? I'd propose syncronise minor version only, while keeping patch for bug fixes. It would also help as follow more semver-like model: when we change the interface of nested-pandas we should change minor versions of both packages.

Keeping the two packages aligned on minor version, but allowing them to differ for bug fixes makes a lot of sense to me. Keeping the two aligned exactly is nice in concept, but can produce a lot of weird semver issues (and package releases that only tick up the dependency version of the other package) as you're rightly pointing out. So this PR will lead to a nested-dask v0.2.0 release.

**Comment by dougbrn** [comment]
the `_meta` property is effectively the empty dataframe, just from the lazy view instead of the eager view. To use an empty dataframe we would need to invoke a head(0) call, which risks some computation and I/O I think? The only advantage of not using `_meta` that I could see is it would allow us to use the "true" dataframe via the eager view, but if `_meta` is not aligned with that then the user has probably already made a workflow-ruining mistake...

**Comment by dougbrn** [comment]
@hombit Here's an example of where I think head(0) is doing some work:

Screen Shot 2024-08-27 at 10 46 28 AM

**Review by hombit — APPROVED** [review]

**Pull request merged**